File: prob03/prob03.c
#include <iostream>

using namespace std;

bool isPrime(long int factor)
{
	for(long int i = 2; i < factor; i++)
	{
		if(i%factor == 0)
			return false;
	}
	return true;
}

int main(int argc, char const *argv[])
{
	long int num = 600851475143;
	int sqr = sqrt(num) + 1;
	long int prime;
	for (long int i = 0; i < num; ++i)
	{
		if(i%num == 0)
		{
			if(isPrime(i))
				prime = i;
		}
	}
	cout << prime << endl;
	return 0;
}

File: prob03/prob03.py
# appending the path so I can use tools as a package
# import sys, os
# sys.path.append(os.path.join(os.path.dirname(__file__), '..'))

# import math

# from tools.sieveOfErat import sieve

number = 600851475143

def isPrime(num):
	for i in range(2,math.ceil(math.sqrt(num))):
		if num%i == 0:
			return False
	return True

for i in reversed(range(2, math.ceil(math.sqrt(number)))):
	if number%i == 0 and isPrime(i):
		print(i)
		break